lib/rack/turnout.rb

```diff
@@ -14,9 +14,14 @@ def initialize(app, config={})
 
   def call(env)
     request = Turnout::Request.new(env)
-    settings = Turnout::MaintenanceFile.find
+    mf = Turnout::MaintenanceConfig.find
+    if mf
+      settings = mf
+    else
+      settings = Turnout::MaintenanceConfig.default
+    end
 
-    if settings && !request.allowed?(settings)
+    if (Turnout::RedisClient.maintenance?(settings) || mf) && !request.allowed?(settings)
```

Comment:
I think I'd prefer we go with either Redis, or a file store, rather than trying to do both at the same time. It would be better for performance and should make for a cleaner implementation, I'd think. Unless there's a good reason to have both that I'm not thinking of.

Reply:
I believe the reasoning was that we wanted to simply append features instead of reworking the structure too much. I do agree that going with just one would be a cleaner implementation.